### ISSUE TITLE
[Feature Request] Weapon switching with number keys / Half-Life style inventory system

### ISSUE THREAD

**Issue by megatog615** [fgsfdsfgs/perfect_dark#365]
Requesting the ability to switch weapons with the number keys like other PC FPS games. Half-Life for example uses a cascading weapon selection menu that, when you press a number it shows all weapons in that category, and subsequent presses changes the selection. This is a useful system for FPS games with more than ten weapons/items. PD would need each weapon/item given a slot number so that it always appears in the menu in the same position, and 'higher priority' items taking precedent. Of course all of this could be configured by the user.

Here's some defaults I came up with. It's not extremely good; this could be workshopped.

1: melee
2: pistols
3: miscellaneous(shotgun, mines, grenades, tranquilizer, crossbow, anything you probably dont want to spend time reloading in a firefight)
4: automatics/smgs/rifles
5: rocket launchers
6: ???

I suggest this system because simply assigning guns to number keys based on the time the item was picked up(the simpler method) can be confusing and hard to get used to.

**Comment by tunbridgep** [comment]
I don't think this is necessary, most missions you only find a handful of items.

One feature I would very much like to see from half life is the last weapon used button, especially if it's clever and knows to not reselect utility or one off items (mines, camspy, etc etc).

**Comment by megatog615** [comment]
>I don't think this is necessary, most missions you only find a handful of items.

It's so that weapons always show up in an expected category and thus have the same hotkey no matter what.

**Comment by tunbridgep** [comment]
The issue with the wheel is that things often appear in seemingly random positions, so maybe that can be addressed

**Comment by throwaway29292** [comment]
I'd absolutely support that. The stock PD wheel menu is cool looking, but impractical to use with mouse.

**Comment by cylonicboom** [comment]
what if we assigned a numpad key to each favorite slot and used that?

**Comment by fgsfdsfgs** [comment]
That would be a bit awkward to use. Plus these days not everyone has a numpad. I think if we do this, then something HL-like with 5-6 categories would probably work the best, which is what OP is suggesting.
